src/sql/base/browser/ui/table/table.ts

```diff
@@ -49,9 +52,15 @@ export class Table<T extends Slick.SlickData> extends Widget implements IDisposa
 	private _onClick = new Emitter<ITableMouseEvent>();
 	public readonly onClick: Event<ITableMouseEvent> = this._onClick.event;
 
+	//my addition
+	private _onHeaderClick = new Emitter<ITableMouseEvent>();
+	public readonly onHeaderClick: Event<ITableMouseEvent> = this._onHeaderClick.event;
+
 	private _onColumnResize = new Emitter<void>();
 	public readonly onColumnResize = this._onColumnResize.event;
 
+	public selection: Slick.Range[] | boolean;
```

Comment:
any reason you need this property on the table class? callers can grab this from the selection model instead and it's a more directly access.

Reply:
This is a leftover from restoreViewState in editDataGridPanel, it will be changed to a set version